app/src/main/java/com/conupods/OutdoorMaps/View/Directions/ModeSelectActivity.java

```diff
@@ -288,6 +292,13 @@ public void run() {
     @Override
     public void onMapReady(GoogleMap googleMap) {
         mMap = googleMap;
-        mMap.moveCamera(CameraUpdateFactory.newLatLng(mOrigin));
+        MarkerOptions destinationMarker = new MarkerOptions();
+        destinationMarker.position(mDestination);
+        destinationMarker.title(toLongName);
+        // Animating to the touched position
+        googleMap.animateCamera(CameraUpdateFactory.newLatLng(mDestination));
+        // Placing a marker on the touched position
+        googleMap.addMarker(destinationMarker);
+        mMap.moveCamera(CameraUpdateFactory.newLatLng(mDestination));
```

Comment:
Setting the camera location to the destination might be weird since most routes normally start from your current location. I don't think it's a big deal though since the user only really uses the navigation page as their guide.

Reply:
yup, tech debt